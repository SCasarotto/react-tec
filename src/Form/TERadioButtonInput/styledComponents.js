import styled from 'styled-components'

export const Input = styled.input`
	cursor: pointer;
	box-sizing: border-box;
`
