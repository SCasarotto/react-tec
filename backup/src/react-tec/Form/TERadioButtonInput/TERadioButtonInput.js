//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Input } from './styledComponents'

const TERadioButtonInput = (props) => {
    const { style, ...rest } = props

    return <Input type="radio" {...rest} />
}

TERadioButtonInput.propTypes = {
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    checked: PropTypes.bool,
    id: PropTypes.string,
}

export default TERadioButtonInput
