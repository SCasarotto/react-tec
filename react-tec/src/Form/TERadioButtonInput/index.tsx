import React from 'react'

import { Input } from './styledComponents'

export interface TERadioButtonInputProps
	extends React.DetailedHTMLProps<
		React.InputHTMLAttributes<HTMLInputElement>,
		HTMLInputElement
	> {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
}
const TERadioButtonInput: React.FC<TERadioButtonInputProps> = (props) => {
	const { className = '', ...rest } = props
	return (
		<Input
			className={`TERadioButtonInput ${className}`}
			{...rest}
			type="radio"
		/>
	)
}

export default TERadioButtonInput
