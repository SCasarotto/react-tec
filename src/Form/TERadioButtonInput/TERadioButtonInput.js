import React from 'react'

import { Input } from './styledComponents'

const TERadioButtonInput = (props) => <Input {...props} type='radio' />

export default TERadioButtonInput
