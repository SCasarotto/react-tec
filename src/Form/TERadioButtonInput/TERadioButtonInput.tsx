import React from 'react'

import { Input } from './styledComponents'

export interface TERadioButtonInputProps extends React.InputHTMLAttributes<HTMLInputElement> {}
export const TERadioButtonInput = React.forwardRef<HTMLInputElement, TERadioButtonInputProps>(
	(props, ref) => {
		const { className = '', ...rest } = props
		return (
			<Input className={`TERadioButtonInput ${className}`} ref={ref} {...rest} type='radio' />
		)
	},
)
