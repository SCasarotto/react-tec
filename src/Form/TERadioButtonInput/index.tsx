import React from 'react'

import { Input } from './styledComponents'

export interface TERadioButtonInputProps
	extends React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement> {
	ref?:
		| ((instance: HTMLInputElement | null) => void)
		| React.RefObject<HTMLInputElement>
		| null
		| undefined
}
export const TERadioButtonInput: React.FC<TERadioButtonInputProps> = React.forwardRef(
	(props, ref) => {
		const { className = '', ...rest } = props
		return (
			<Input className={`TERadioButtonInput ${className}`} ref={ref} {...rest} type='radio' />
		)
	},
)
