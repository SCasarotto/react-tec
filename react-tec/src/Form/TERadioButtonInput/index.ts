import TERadioButtonInput from './TERadioButtonInput'

export default TERadioButtonInput
