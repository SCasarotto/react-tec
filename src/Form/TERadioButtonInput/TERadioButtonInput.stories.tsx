import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TERadioButtonInput, TERadioButtonInputProps } from './TERadioButtonInput'

export default {
	title: 'Form/TERadioButtonInput',
	component: TERadioButtonInput,
	argTypes: {},
} as Meta

const Template: Story<TERadioButtonInputProps> = (args) => <TERadioButtonInput {...args} />

export const Default = Template.bind({})
Default.args = {}
