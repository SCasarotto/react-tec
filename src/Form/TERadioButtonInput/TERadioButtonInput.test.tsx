import React from 'react'
import { render } from '@testing-library/react'
import { TERadioButtonInput } from './TERadioButtonInput'

describe('TERadioButtonInput', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(<TERadioButtonInput />)
		expect(asFragment()).toMatchSnapshot()
	})
})
