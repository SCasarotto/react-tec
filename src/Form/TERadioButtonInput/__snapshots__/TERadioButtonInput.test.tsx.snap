// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TERadioButtonInput matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="sc-bdnylx dpqFiO TERadioButtonInput "
    type="radio"
  />
</DocumentFragment>
`;
